renku/core/dataset/dataset.py

```diff
@@ -1315,3 +1321,60 @@ def store_dataset_data_location(dataset: Dataset, location: Optional[Path]) -> N
 def read_dataset_data_location(dataset: Dataset) -> Optional[str]:
     """Read data location for a dataset in the config file."""
     return get_client().get_value(section="dataset-locations", key=dataset.name, config_filter=ConfigFilter.LOCAL_ONLY)
+
+
+@inject.autoparams("client_dispatcher", "storage_factory")
+def mount_external_storage(
+    name: str,
+    existing: Optional[Path],
+    unmount: bool,
+    yes: bool,
+    client_dispatcher: IClientDispatcher,
+    storage_factory: IStorageFactory,
+) -> None:
+    """Mount an external storage to a dataset's data directory.
+
+    Args:
+        name(str): Name of the dataset
+        existing(Optional[Path]): An existing mount point to use instead of actually mounting the external storage.
+        unmount(bool): Unmount the external storage.
+        yes(bool): Don't prompt when removing non-empty dataset's data directory.
+    """
+    client = client_dispatcher.current_client
+    datasets_provenance = DatasetsProvenance()
+
+    dataset = datasets_provenance.get_by_name(name=name, strict=True)
+
+    if not dataset.storage and not unmount:
+        communication.warn(f"Cannot mount dataset '{name}': Dataset doesn't have a storage backend")
+        return
+
+    datadir = client.path / dataset.get_datadir()
+
+    # NOTE: Try to unmount the path in case it was mounted before
+    unmount_path(datadir)
+
+    if unmount:
```

Comment:
it's a bit weird that `mount_external_storage(...)` does the exact opposite of its name if `unmount` is set true. Would be nicer if it's two separate methods (and `mount_external_storage` can call `unmount_external_storage`)